nomad/structs/structs_test.go

```diff
@@ -617,6 +617,62 @@ func TestJob_RequiredSignals(t *testing.T) {
 	}
 }
 
+// test new Equal comparisons for components of Jobs
+func TestJob_PartEqual(t *testing.T) {
+	ns := &Networks{}
+	if !ns.Equals(&Networks{}) {
+		t.Fatal("network set wise not equal")
+	}
+
+	ns = &Networks{
+		&NetworkResource{Device: "eth0"},
+	}
+	if !ns.Equals(&Networks{
+		&NetworkResource{Device: "eth0"},
+	}) {
```

Comment:
I would encourage use of `testify` as it has nicer output than plain `t.Fatal()` that makes it easier to find failing conditions in a stream of logs and test output.

This would be something like:

```go
require.True(t, ns.Equals(&Networks{&NetworkResource{Device: "eth0"}}))
```